command/command.go

```diff
@@ -152,3 +165,15 @@ func ExecuteCommand(ctx context.Context, execution *Execution) (*Execution, erro
 
 	return execution, nil
 }
+
+func escapeZombie(ex *Execution) {
+	logger := logrus.WithFields(logrus.Fields{"component": "command"})
+	if ex.InProgress != nil && ex.InProgressMu != nil && ex.Name != "" {
+		logger.WithField("check", ex.Name).Warn("check or hook execution created zombie process - escaping in order for the check to execute again")
+		ex.InProgressMu.Lock()
+		delete(ex.InProgress, ex.Name)
```

Comment:
I was going to ask where the `InProgress` map was being added, but I found it in `executeHook` Looks good!

LGTM!